### ISSUE TITLE
Text too long; german translation

### ISSUE THREAD

**Issue by merlinuwe** [syncthing/syncthing#3702]
Hi, 
thank you for implementing the global and local state of the folders. Now one can easiely see, if they are synced or not. But when there are lots of files, the text isn't readable completely (see screenshot).
The same issue is with "last change".

For translation into german language: 
files = Dateien
directories = "Verzeichnisse" or "Ordner" 

![text_too_long](https://cloud.githubusercontent.com/assets/4138869/19960601/2bc9be0a-a1ae-11e6-864e-4b7865d33e9b.jpg)

---------
The button for changing the language is lettered in english words. I think, it should be written in the language of the user. For example, "German" should be "Deutsch".

Here is a list of languages which could help the translator:
https://meta.wikimedia.org/wiki/List_of_Wikipedias

Thank you for your great piece of software.

Greetings,

merlinuwe

### Version Information

Syncthing Version: v0.14.10
OS Version: Windows 10
Browser Version: Firefox 49.0.2, Chrome

**Comment by canton7** [comment]
I *think* there are 3 separate issues here?

1. But when there are lots of files, the text isn't readable completely (see screenshot).
2. "files" and "directories" aren't yet translated.
3. Language selection should use the language of the user


But when there are lots of files, the text isn't readable completely (see screenshot).
----

Probably needs some work.


"files" and "directories" aren't yet translated.
----

Indeed they aren't. Please provide translations [on Transifex](https://www.transifex.com/syncthing/syncthing/translate/#en_GB/gui/98380552?translated=no).

Language selection should use the language of the user
-----

Obviously this isn't possible - we don't know what the language of the user is until they've selected their language. We could display each language using its local name though.

**Comment by wweich** [comment]
> Obviously this isn't possible - we don't know what the language of the user is until they've selected their language. We could display each language using its local name though.

We should (maybe) have each language name in that language instead of English.  So instead of 
English, German, French, ...
we would have
English, Deutsch, Français, ...

If a German user cannot speak any English, he may not know that he has to select "German".

**Comment by canton7** [comment]
> We should (maybe) have each language name in that language instead of English. So instead of
English, German, French, ...
we would have
English, Deutsch, Français, ...

> If a German user cannot speak any English, he may not know that he has to select "German".

Yep, that's what I meant by "We could display each language using its local name though.".

**Comment by merlinuwe** [comment]
Yes, please display each language using its local name.
I joined transiflex a few minutes ago.

**Comment by merlinuwe** [comment]
ups.

**Comment by calmh** [comment]
Yeah. We use the Transifex API to list the available languages, and so far (when I looked last) it only provides the language names in English. I agree that this is stupid. If it's fixed, we should use the localized language names. I'm using this issue for that, and other issues can be opened for other stuff.

**Comment by uok** [comment]
> But when there are lots of files, the text isn't readable completely (see screenshot).
As a fix we could add tooltips or create a second line. Any other changes might cause a lot of work as the `text-overflow: ellipsis` is part of the bootstrap framework

**Comment by canton7** [comment]
> But when there are lots of files, the text isn't readable completely (see screenshot).

There's a new issue, #3710, which tracks this.

**Comment by calmh** [comment]
(FWIW I actually implemented this in a branch, but some of the language names are rather long in their localized versions and it mostly broke the GUI.)

**Comment by uok** [comment]
We could use the word "language" (translated in current language) as menu item and a modal for selection

**Comment by superuser-does** [comment]
It's great to hear it's been investigated.

I understand you have native language names to hand, but if you don't, I would look no further than the well-curated [list in Wikipedia's language selector](https://github.com/wikimedia/jquery.uls/blob/master/src/jquery.uls.data.js) widget.

The list (JS file) is fairly easy to parse and has corresponding ISO codes, or you can just take language names manually as required.

With respect to the widget; I would encourage you to consider a simple HTML select box, @calmh. It's not a control the user will really be interacting with much, so you don't have to have a pretty display of all the languages at once. As soon as they find their language, they'll be good to go IMO. Hope this helps!
